src/main/java/com/amihaiemil/zold/WaitForWallet.java

```diff
@@ -109,13 +109,14 @@ public Wallet handleResponse(final HttpResponse httpResponse)
                                 + TIME_TO_WAIT.toMinutes() + " minutes");
                     }
                 } while (true);
-
-                return new RtWallet(this.client, this.baseUri);
-            } catch (final InterruptedException | URISyntaxException ex) {
+            } catch (final URISyntaxException ex) {
                 throw new IllegalStateException(
                         "Exception while waiting for Wallet!", ex
                 );
+            } catch (final InterruptedException iex) {
+                Thread.currentThread().interrupt();
```

Comment:
@paulodamaso after interrupting the Thread, shouldn't we throw IllegalStateException as well? :D

Reply:
@amihaiemil According to the ticket description and the  documentation provided we must choose between one of two: either rethrow the exception or interrupt the thread